pkg/infrastructure/services/messaging_kafka.go

```diff
@@ -58,7 +58,7 @@ func (k *kafkaMessagingDeployer) createRequiredKafkaTopics(infra *v1alpha1.Kogit
 		}
 	}
 	// topics required by the deployed service
-	topics, err := k.fetchRequiredTopics(service)
+	topics, err := k.fetchTopicsAndSetCloudEventsStatus(service)
```

Comment:
Generally we update `KogitoService` status field in `defer updateStatus` function. Can we call update `CloudEventsStatus` from that function to maintain uniformatity in code

Reply:
That's not entirely true. We were not implementing the Status pattern correctly TBH. There's more work to be done in this field. `updateStatus` function is to get the general status information about the resources we manage after the reconcile loop, does not mean that all status update should be in there, a renaming could work to clearly express its meaning. 

Fetching the topics again or carry some kind of context through the reconcile cycle just to only update the status spec in one function does not make much sense now.